app/javascript/packages/document-capture/services/upload.tsx

```diff
@@ -77,38 +72,35 @@ async function upload(payload, { method = 'POST', endpoint, csrf }) {
     return new Promise(() => {});
   }
 
-  const result = /** @type {UploadSuccessResponse|UploadErrorResponse} */ (await response.json());
+  const result = (await response.json()) as UploadSuccessResponse | UploadErrorResponse;
```

Comment:
This is another one we can probably avoid casting in favor of type-ing the variable itself:

```suggestion
  const result: UploadSuccessResponse | UploadErrorResponse = await response.json();
```